libs/@local/harpc/service/src/metadata.rs

```diff
@@ -0,0 +1,34 @@
+//! Version metadata for both procedures and services.
+
+use harpc_types::version::Version;
+
+/// Represents the deprecation information for a procedure or service.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct Deprecation {
+    /// The version at which the procedure/service was deprecated.
+    pub since: Version,
+    /// The reason for deprecation.
+    pub reason: Option<&'static str>,
+}
+
+/// Represents the removal information for a procedure or service.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct Removal {
+    /// The version at which the procedure/service will be removed.
+    pub version: Version,
+    /// The reason for removal.
+    pub reason: Option<&'static str>,
+}
```

Comment:
I don't really understand this. How I can get information for a removed procedure?

Reply:
It's about _when_ something is scheduled for removal, like on v6.0 this will be removed.

Reply:
I don't think this is needed. A deprecation warning should be good enough for that. If a function is stable in `v3`, then it's deprecated in `v4`, then it's expected to be removed later.

Reply:
you are right, done in [6551827](https://github.com/hashintel/hash/pull/5283/commits/6551827b6ccb2654673dfc6f1b8bf7ee29942222)